web/v1/redshift/sql-runner/sql/custom/01-indicative-export/03-indicative-export-staged.sql

```diff
@@ -0,0 +1,281 @@
+     ON dr.root_id = b.event_id
+     AND dr.root_tstamp = b.collector_tstamp
+
+  user_mapping AS
+    (SELECT
+       domain_userid,
+       user_id AS custom_user_id,
+       MIN(derived_tstamp) AS first_seen_tstamp
+     FROM {{.scratch_schema}}.user_stitching{{.entropy}}
+     GROUP BY 1,2)
+SELECT
+  c.app_id,
+  c.br_family,
+  c.br_features_director,
+  c.br_features_flash,
+  c.br_features_gears,
+  c.br_features_java,
+  c.br_features_pdf,
+  c.br_features_quicktime,
+  c.br_features_realplayer,
+  c.br_features_silverlight,
+  c.br_features_windowsmedia,
+  c.br_lang,
+  c.br_name,
+  c.br_renderengine,
+  c.br_type,
+  c.br_version,
+  c.br_viewheight,
+  c.br_viewwidth,
+  c.doc_height,
+  c.doc_width,
+  c.domain_sessionid,
+  c.domain_sessionidx,
+  c.domain_userid,
+  c.dvce_ismobile,
+  c.dvce_screenheight,
+  c.dvce_screenwidth,
+  c.dvce_type,
+  c.event,
+  c.event_id,
+  c.event_vendor,
+  c.geo_city,
+  c.geo_country,
+  c.geo_region,
+  c.geo_region_name,
+  c.geo_timezone,
+  c.geo_zipcode,
+  c.mkt_campaign,
+  c.mkt_clickid,
+  c.mkt_content,
+  c.mkt_medium,
+  c.mkt_network,
+  c.mkt_source,
+  c.mkt_term,
+  c.network_userid,
+  c.os_family,
+  c.os_manufacturer,
+  c.os_name,
+  c.os_timezone,
+  c.page_referrer,
+  c.page_title,
+  c.page_url,
+  c.page_urlfragment,
+  c.page_urlhost,
+  c.page_urlpath,
+  c.page_urlquery,
+  c.platform,
+  c.refr_domain_userid,
+  c.refr_medium,
+  c.refr_source,
+  c.refr_term,
+  c.refr_urlfragment,
+  c.refr_urlhost,
+  c.refr_urlpath,
+  c.refr_urlquery,
+  c.se_action,
+  c.se_category,
+  c.se_label,
+  c.se_property,
+  c.se_value,
+  c.ti_category,
+  c.ti_currency,
+  c.ti_name,
+  c.ti_orderid,
+  c.ti_price,
+  c.ti_price_base,
+  c.ti_quantity,
+  c.ti_sku,
+  c.tr_affiliation,
+  c.tr_city,
+  c.tr_country,
+  c.tr_currency,
+  c.tr_orderid,
+  c.tr_shipping,
+  c.tr_shipping_base,
+  c.tr_state,
+  c.tr_tax,
+  c.tr_tax_base,
+  c.tr_total,
+  c.tr_total_base,
+  c.user_fingerprint,
+  c.user_id,
+  c.user_ipaddress,
+  c.useragent,
+  c.collector_tstamp,
+  c.event_name,
+  c.custom_event_name,
+  c.derived_tstamp,
+  c.change_form_form_id,
+  c.change_form_element_id,
+  c.change_form_node_name,
+  c.change_form_element_classes,
+  c.submit_form_form_id,
+  c.submit_form_form_classes,
+  c.submit_form_elements,
+  c.link_click_element_id,
+  c.link_click_element_classes,
+  c.link_click_element_target,
+  c.link_click_target_url,
+  c.sign_up_email,
+  c.sign_up_company,
+  c.sign_up_service_type,
+  c.demo_request_company,
+  c.demo_request_email,
+  c.demo_request_insights,
+  c.page_view_id,
+  CASE
+    WHEN c.user_id IS NOT NULL THEN c.user_id
+    WHEN u.custom_user_id IS NULL THEN c.domain_userid
+    WHEN c.user_id IS NULL THEN u.custom_user_id
+  END AS alias_user_id
+FROM child_table_joins AS c
+LEFT JOIN user_mapping AS u
+ON u.domain_userid = c.domain_userid
+AND u.first_seen_tstamp < c.derived_tstamp);
```

Comment:
Not sure we need this join on dates as well but again let us figure out the answer to Colm's Q and proceed from there